pkg/epp/handlers/request.go

```diff
@@ -32,6 +32,37 @@ import (
 func (s *StreamingServer) HandleRequestHeaders(reqCtx *RequestContext, req *extProcPb.ProcessingRequest_RequestHeaders) error {
 	reqCtx.RequestReceivedTimestamp = time.Now()
 
+	// Headers must be processed first to populate the request context as subsequent logic (like body processing) may
```

Comment:
I am not following, where in this function do we use the reqCtx that would require moving this logic earlier? it is fine to move it up, but wanted to double check where the bug is in this specific case.